components/mediation-config-admin/org.wso2.carbon.mediation.configadmin/src/main/java/org/wso2/carbon/mediation/configadmin/ConfigurationUpdater.java

```diff
@@ -373,8 +373,26 @@ private void setFileNames(SynapseConfiguration newConfig) {
                 newEndpoint.setArtifactContainerName(oldEndpoint.getArtifactContainerName());
                 addToDeploymentStore(MultiXMLConfigurationBuilder.ENDPOINTS_DIR, oldEndpoint.getFileName(), name, newConfig);
             } else {
-                newEndpoint.setFileName(name + XML);
-                addToDeploymentStore(MultiXMLConfigurationBuilder.ENDPOINTS_DIR, newEndpoint.getFileName(), name, newConfig);
+                if (currentConfig.getDefinedEndpoints().size() >= newConfig.getDefinedEndpoints().size()) {
+                    newConfig.removeEndpoint(name);
+                    log.error("Unable to update the endpoints.");
```

Comment:
We should mention the reason on why we can't update the endpoints. Otherwise, a person who looks at the log has no idea why it was unable to update the ep.